### PR TITLE
BrownDog conversion out of met.process into convert.input

### PR DESCRIPTION
**Pull request by bcow**
The brown dog hack in met.process has been commented out (though could be brought back if everything else turns out to be broken)

The browndog list (which currently does NOT have username and password) is now passed to convert.inputs where conversions can now be done on
1) The localhost
2) Remote server
3) Brown Dog - with downloading and additional functions like metgapfill on localhost.*
*This won't work anymore because now the DAP requires a username and password - once this is working I'll add it to the pull request.

Output and input types are currently defined by lookup tables but ultimately will be fields in bety.

### CODE REVIEW

**Comment by mdietze** [comment]
@bcow I'm a bit confused on the NARR part of this workflow, which doesn't seem to be what we discussed. Is this commit just intended to get Ameriflux working?  For NARR it appears that we're downloading NARR locally even if we're using browndog (there's no browndog check on the download). Second, the NARR workflow doesn't appear to be correct even if we're running locally as the code to do the CF conversion, permuting, and rechunking appears to be absent and instead you've just hard-coded the ID for the converted+permuted version. Thus, if someone else was running this they'd download the NARR but never convert it? Or am I missing something?  I'm accepting this pull now, since Ameriflux is the priority, but please touch base with me about what's going on.

**Comment by mdietze** [comment]
@bcow Also, there's now conflicts between the mainline and this pull because of @dlebauer 's commit so I'll need you to resolve those.

**Comment by bcow** [comment]
No, changes to NARR haven't been committed - I was focusing on getting Ameriflux working and then saving NARR for when we can sit down and write wrappers for NARR and get it downloaded and prepped on Brown Dog.

**Comment by bcow** [comment]
The question about hardcoding in the database IDs for raw and CF NARR is one I don't have a solution to. I haven't automated that process because right now it involves running scripts on the cluster which is unrealistic for any users who don't have access to our/ any computing cluster.

**Comment by mdietze** [comment]
@bcow, my specific question wasn't about hard coding in the ID for the raw NARR, but hardcoding the fully processed ID and skipping all the processing steps. That said, I don't see why you would need to run scripts on a remote server to figure out what row in the Inputs database contains a specific data set -- that should be something that just requires the database.

More generally, we need to get to the point where this workflow will work when we add a 3rd, 4th, and 5th met product. And by that I mean that we can't expect users to edit the source code and hardcode in new special cases for every product. I don't think that generalization needs to be in this specific commit, but please stop by so we can talk through a plan.

**Comment by bcow** [comment]
Current status of the met workflow:
Green is working, 
red has a problem, 
blue still needs needs to be implemented.

I believe @robkooper has said he will work on getting NARR cached on Brown Dog. 
We can divvy up the rest of the work - I think I could help with writing the wrappers for SIPNET and LINKAGES. 

screen shot 2015-04-24 at 6 45 00 pm

**Comment by dlebauer** [comment]
It would be great to provide such a table w/ links to the implementation
and any tests/examples so that, eg, I can implement the biocro wrapper (or
is this the existing met2model functions?)

Also, what distinguishes 'pecan2' from 'PEcAn'?

**Comment by mdietze** [comment]
pecan2 = pecan2.bu.edu
geo = geo.bu.edu

Yes David, we think it would be best if you the one who implements the biocro met functions.  Rob or Edgar could probably help you with their brown dog wrappers 

<iPhone>

File: utils/R/convert.input.R [hunk 1]
```diff
@@ -18,14 +18,14 @@ convert.input <- function(input.id,outfolder,formatname,mimetype,site.id,start_d
...
-  print("start CHECK")
-  check = dbfile.input.check(site.id, startdate, enddate, mimetype, formatname, parentid=input.id, con=con, hostname)
-  print("end CHECK")
-  print(check)
-  if(length(check)>0){
-    return(check$container_id)
-  }
-  
+  #   print("start CHECK")
+  #   check = dbfile.input.check(site.id, startdate, enddate, mimetype, formatname, parentid=input.id, con=con, hostname)
```
Comment (mdietze) [inline utils/R/convert.input.R]: Why is the CHECK disabled in convert.inputs?

File: utils/R/convert.input.R [hunk 2]
```diff
@@ -6,7 +6,7 @@
 ##' @export
 ##' @author Betsy Cowdery, Michael Dietze
 convert.input <- function(input.id,outfolder,formatname,mimetype,site.id,start_date,end_date,
-                          pkg,fcn,username,con=con,hostname='localhost',write=TRUE,...){
+                          pkg,fcn,username,con=con,hostname='localhost',browndog, write=TRUE,...){
   print(paste("Convert.Inputs",fcn,input.id,hostname))
```
Comment (dlebauer) [inline utils/R/convert.input.R]: Is there a reason to use `print` instead of the `logger.*` functions here?

File: utils/R/convert.input.R (hunk 1 above)
Comment (bcow) [inline utils/R/convert.input.R]: Good catch - that was commented out for a test. I'll change it back.

File: utils/R/convert.input.R (hunk 2 above)
Comment (bcow) [inline utils/R/convert.input.R]: As far as I know, the logger functions are actually what I should be using. I'm not sure how much of the printed output we want to keep - I know I need to ultimately go back and remove some of the sanity check printouts. @mdietze @robkooper should I change these to using logger functions?

**Comment by mdietze** [comment]
Status update: I talked to Betsy and I'm going to Merge this pull so that we can collectively put it through it's paces for testing/validation for the next release. Betsy will then clean up debugging prints and other small issues when she's updating the documentation and tests, and that will come as as separate pull (along with fixes to any bugs we all find).

**Pull request merged**